### PR TITLE
test(rendering): the .ply Gaussian-splat reader was graded only where sim-gs was installed

### PR DESCRIPTION
**Pull request by cagataycali**
## The reader was graded only where a CUDA rasterizer was already installed

`_load_ply_splats` reads the standard 3DGS `.ply` backdrop -- the format every
`GSPLAT_SCENES` preset ships as, and what Nerfstudio, Polycam and Marble export.
Its six cells each opened with `pytest.importorskip("plyfile")`. `plyfile` is
declared only in `sim-gs`, which `[all]` does not pull, so on a `[all,dev]` tree
they report as skips and the decode is measured by nothing:

| `pytest tests/rendering` | reader cells run | `backgrounds.py` |
|---|---|---|
| before | **0** of 6 (6 skipped) | 90%, 47 lines missing |
| after | **9** of 11 (2 skipped) | **97%**, 15 lines missing |

The 15 that remain are `gsplat_rasterizer_available`'s probe body, which needs a
real CUDA `gsplat` and cannot be reached any other way.

## What the reader decides, and what it now costs to get wrong

reader contracts

Every number above is read back out of `_load_ply_splats`, not restated.

| decision | wrong-but-plausible alternative | how it presents |
|---|---|---|
| `f_rest_{c*(K-1)+j}` is channel `c`, coefficient `j` | read the block coefficient-major | view-dependent colour moves the wrong way |
| `scale_*` is log-space, `opacity` is a logit | forward the stored numbers | sub-millimetre gaussians at ~50% alpha: a haze |
| DC-only -> `(N, 3)` baked RGB; SH -> `(N, K, 3)` raw | emit a degenerate `(N, 1, 3)` | `render` derives `sh_degree` from `shape[1]` and evaluates the wrong basis |
| a `.ply` carries no antialiased flag | inherit `.spz`'s header bit | AA opacity compensation applied to an asset not trained with it |

Mutating the reader four ways, each mutation reds exactly the cell that owns it:

| mutation | result |
|---|---|
| drop both activations | 1 failed / 8 passed |
| `reshape(-1, n_rest, 3)` (coefficient-major) | 3 failed / 6 passed (one per degree) |
| add `"antialiased": True` to the return | 2 failed / 7 passed (reader **and** consumer) |
| neuter the degree check | 1 failed / 8 passed |

## The seam

The cells write a real `binary_little_endian` 3DGS PLY -- including the
`nx ny nz` normals INRIA writes and this reader ignores, so a positional read of
the property block would not survive -- and stand in the one thing `plyfile`
does for this caller: hand back the vertex element as a structured array with
one field per declared property.

A double is worth its faithfulness, so that is a cell too.
`test_the_stand_in_reads_what_plyfile_reads` parses the same bytes both ways and
compares column by column, over both property blocks. It is the only cell that
needs the extra, and it is why the other nine do not. Dropping one property from
the stand-in reds it on both parametrisations.

```
plyfile absent   838 passed, 5 skipped      <- what [all,dev] runs
plyfile present  11 passed                  <- includes the faithfulness cell
```

## Three contracts that had no cell at all

* a `.ply` reaches the PLY reader and leaves `_rasterize_mode` at `classic`
  (suffix dispatch plus the antialiased-flag absence, read through
  `GsplatBackground._load` rather than off the reader's dict);
* `device="cuda"` against a torch without CUDA is refused **before** the scene
  is read (`_splats is None`), which is the documented fail-loud-and-early;
* an unreadable scene raises `PermissionError` at construction -- a documented
  `Raises:` entry with nothing behind it.

## Net

| | lines |
|---|---|
| `tests/rendering/test_ply_gaussian_splat_reader.py` | +365 |
| `tests/rendering/test_backgrounds.py` | +4 / **-185** |
| net | **+184** |

The deletion is the six unreachable cells plus their `_build_ply` helper. Each
fact they pinned is pinned in the new file -- key set, means and quaternion
passthrough, `exp` scales, `sigmoid` opacity, the DC bake and its clip, the
saturating case, raw SH with DC first, the channel-major mapping, the DC-only
`(N, 3)` shape, and both refusals -- several of them wider (degrees 1, 2 and 3
where there was one, and a value that encodes its own `(channel, coefficient)`
so a transpose is legible rather than merely unequal). `test_backgrounds.py`
keeps a four-line pointer to where the contract lives.

Test-only; no library behaviour is modified.

## Checks

`ruff check` + `ruff format --check` clean over `strands_robots tests tests_integ`.
`mypy` 20 errors in 6 files, identical to `upstream/main`.
`scripts/check_whole_tree_graders.py` 297 passed.
Full `MUJOCO_GL=egl pytest tests` 59 failed / 45023 passed / 54 errors; the 113
failing ids are set-identical to the same list run on a pristine `main`
worktree, 0 attributable.

### CODE REVIEW

**Review by yinsong1986 — APPROVED** [review]
## Summary

Test-only PR that moves the `.ply` 3DGS reader contract out from behind `pytest.importorskip("plyfile")` -- which only ever ran where the GPU-only `sim-gs` extra was installed -- into a new `tests/rendering/test_ply_gaussian_splat_reader.py` that writes real `binary_little_endian` INRIA-layout bytes and stands in the one thing `plyfile` does for this caller (structured-array access to the vertex element). I verified the new assertions against the shipped `_load_ply_splats` on `main`: the channel-major `f_rest` mapping matches the `reshape(-1, 3, n_rest).transpose(0, 2, 1)` in the reader, the exp/sigmoid activations, the DC bake and its clip, both `ValueError` refusals' messages, the classic `_rasterize_mode` for suffix-dispatched PLYs, and the CUDA-refused-before-decode ordering in `_load` are all pinned faithfully. The two fixture seams (`sys.modules` and `strands_robots.utils._lazy_modules`) are exactly the pair the reader consults (`require_optional` then `from plyfile import PlyData`), both patched via `monkeypatch.setitem` per AGENTS.md, and the faithfulness cell (`test_the_stand_in_reads_what_plyfile_reads`) keeps the stand-in honest against the real dependency where it is installed. Every fact the six deleted `test_backgrounds.py` cells pinned reappears in the new file, several widened (degrees 1-3, a transpose-legible SH encoding, normals in the property block so a positional read cannot pass). No library behaviour is modified.

## What's good

- The stand-in is graded against the real `plyfile` over both property-block shapes, so the double cannot silently diverge from the dependency it replaces.
- `monkeypatch.setitem(sys.modules, ...)` for both injected modules, per AGENTS.md's orphaned-module rule -- no bare `sys.modules` mutation.
- The mutation table in the PR description demonstrates each cell reds exactly the decision it owns.
- `test_backgrounds.py` keeps a pointer comment to where the contract now lives, so the next reader of the `.spz` cells finds the `.ply` siblings.
